### PR TITLE
feat: lists react to value prop changes in items

### PR DESCRIPTION
**Pull request by pr3tori4n**
**Related Issue:** #738 

## Summary

This addresses the bug Kumar raised in which if the value of an item changed after that item was selected, the parent list's Map was not updating its keys. This is due to the mutationObserver not reacting to attribute changes - which is necessary since attribute changes are too frequent.
So we added a watcher to the value prop and emit a custom event when it changes, that the list can listen to and respond accordingly.

### CODE REVIEW

File: src/components/calcite-pick-list-item/calcite-pick-list-item.tsx [hunk 1]
```diff
@@ -127,6 +131,13 @@ export class CalcitePickListItem {
    */
   @Event() calciteListItemPropsUpdated: EventEmitter;
 
+  /**
+   * Emitted whenever the the item's value property is modified.
+   * @event calciteListItemPropsUpdated
```
Comment (jcfranco) [inline src/components/calcite-pick-list-item/calcite-pick-list-item.tsx]: The event name doesn't match.

File: src/components/calcite-pick-list-item/calcite-pick-list-item.tsx [hunk 2]
```diff
@@ -98,6 +98,10 @@ export class CalcitePickListItem {
    */
   @Prop({ reflect: true }) value!: string;
 
+  @Watch("value") valueWatchHandler(newValue, oldValue) {
+    this.calciteListItemValueUpdated.emit({ oldValue, newValue });
```
Comment (driskull) [inline src/components/calcite-pick-list-item/calcite-pick-list-item.tsx]: Should this be named `calciteListItemValueChange`? Events shouldn't be in past tense and we've been using `change` for events.

File: src/components/calcite-pick-list-item/calcite-pick-list-item.tsx (hunk 1 above)
Comment (driskull) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.tsx]: Should this be named calciteListItemPropsChange? Events shouldn't be in past tense and we've been using change for events.

**Comment by pr3tori4n** [comment]
@driskull can we make the existing event name change in a separate PR?

**Comment by driskull** [comment]
Yes, it can be a separate pr for `calciteListItemPropsUpdated` but its a breaking change so we can either do it now or deprecate later.

**Comment by pr3tori4n** [comment]
@driskull I added a test. Please take a second look.

**Comment by pr3tori4n** [comment]
> 👍
> Can you link the issue for renaming the other event to this review?

I don't believe there is an issue for it. Mind creating one?

File: src/demos/value-list/valueUpdate.html [hunk 3]
```diff
@@ -0,0 +1,57 @@
+<!DOCTYPE html>
```
Comment (jcfranco) [inline src/demos/value-list/valueUpdate.html]: For consistency, the demo filename should be kebab-cased.

**Comment by driskull** [comment]
https://github.com/Esri/calcite-app-components/issues/773

**Pull request merged**